main.c

```diff
@@ -686,13 +686,30 @@ int main(void)
 	gpio_set_dir(PICO_DEFAULT_LED_PIN, GPIO_OUT);
 	gpio_put(PICO_DEFAULT_LED_PIN, 1);
 
-	gpio_init(BOOTLOADER_ENTRY_PIN);
-	gpio_pull_up(BOOTLOADER_ENTRY_PIN);
-	gpio_set_dir(BOOTLOADER_ENTRY_PIN, 0);
+	struct image_header *hdr = (struct image_header *)(XIP_BASE + IMAGE_HEADER_OFFSET);
+
+	gpio_set_function(UART_TX_PIN, GPIO_FUNC_UART);
+	gpio_set_function(UART_RX_PIN, GPIO_FUNC_UART);
+	uart_init(uart0, UART_BAUD);
+	uart_set_hw_flow(uart0, false, false);
```

Comment:
you shouldn't use uart0, you should use a define instead

Reply:
definition replaced in the code